src/core/infer_request.cc

```diff
@@ -110,6 +113,71 @@ InferenceRequest::SetPriority(uint32_t p)
   }
 }
 
+#ifdef TRITON_ENABLE_TRACING
+Status
+InferenceRequest::TraceTensor(const std::string& msg)
```

Comment:
It is ok to have "msg" for logging, but you also need an argument of type TRITONSERVER_InferenceTraceActivity. Those values uniquely identify what the trace record represents. Right now we have TRITONSERVER_TRACE_TENSOR_INPUT and TRITONSERVER_TRACE_TENSOR_OUTPUT. But it seems like instead we should have:
TRITONSERVER_TRACE_TENSOR_QUEUE_INPUT
TRITONSERVER_TRACE_TENSOR_BACKEND_INPUT
TRITONSERVER_TRACE_TENSOR_BACKEND_OUTPUT